src/book/book.controller.ts

```diff
@@ -14,6 +16,38 @@ import { BookDto } from './dto'
 export class BookController {
   constructor(private readonly bookService: BookService) {}
 
+  @Get()
+  async getAllBooks(@Query() query: any) {
+    const { page, limit, ...filters } = query
+    if (page && isNaN(parseInt(page))) {
+      throw new NotFoundException('Invalid page number')
+    }
+    if (limit && isNaN(parseInt(limit))) {
+      throw new NotFoundException('Invalid limit')
+    }
+    const books = await this.bookService.getAllBooks(
+      filters,
+      page ? parseInt(page) : 1,
+      limit ? parseInt(limit) : 20,
+    )
+    if (!books || books.total === 0) {
+      throw new NotFoundException('No books found')
+    }
+    return books
+  }
+
+  @Get(':id')
+  async getBookById(@Param('id') id: string) {
```

Comment:
I would like to have a getBookByISBN or something like that too (if it's possible)